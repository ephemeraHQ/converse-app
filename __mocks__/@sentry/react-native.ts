const init = jest.fn();
const captureException = jest.fn();
const captureMessage = jest.fn();
const addBreadcrumb = jest.fn();
const withScope = jest.fn();

const MockedSentry = {
  init,
  captureException,
  captureMessage,
  addBreadcrumb,
  withScope,
};

export { init, captureException, captureMessage, addBreadcrumb, withScope };

export default MockedSentry;
