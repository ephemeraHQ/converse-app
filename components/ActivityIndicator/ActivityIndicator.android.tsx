import { ActivityIndicator } from "react-native-paper";

export default ActivityIndicator;
