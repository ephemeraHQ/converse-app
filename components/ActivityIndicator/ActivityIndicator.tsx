import { ActivityIndicator } from "react-native";

export default ActivityIndicator;
