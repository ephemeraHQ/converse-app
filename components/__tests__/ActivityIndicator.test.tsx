import { render } from "@testing-library/react-native";
import React from "react";
import { Platform } from "react-native";

import ActivityIndicator from "../ActivityIndicator/ActivityIndicator"; // Import the default platform-specific component

// Snapshot test for the Android-specific ActivityIndicator component
describe("ActivityIndicator Component", () => {
  it("renders correctly on Android", () => {
    // Temporarily set the platform to Android
    Platform.OS = "android";

    const tree = render(<ActivityIndicator />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("renders correctly on iOS/Default", () => {
    // Temporarily set the platform to iOS or default
    Platform.OS = "ios";

    const tree = render(<ActivityIndicator />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
