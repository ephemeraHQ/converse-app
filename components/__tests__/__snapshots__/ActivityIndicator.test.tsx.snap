// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ActivityIndicator Component renders correctly on Android 1`] = `<ActivityIndicator />`;

exports[`ActivityIndicator Component renders correctly on iOS/Default 1`] = `<ActivityIndicator />`;
